### ISSUE TITLE
Script crash during dataset preparation

### ISSUE THREAD

**Issue by ADetilie** [hovsg/HOV-SG#5]
### TL;DR

I am getting the following error during Habitat Matterport dataset prep:
```bash
-------------
{}
agent_state: position [13.376938    0.09943584  0.18875428] rotation quaternion(1, 0, 0, 0)
saving frame 3584/3586: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3585/3585 [03:05<00:00, 19.28it/s]
GL::Context::current(): no current context
[1]    234262 IOT instruction (core dumped)  
```

### Steps to reproduce:
1. I've downloaded "Downloading HM3D v0.2" from https://github.com/matterport/habitat-matterport-3dresearch
and prepared it as described in the `README.md`.

See the image:
![image](https://github.com/user-attachments/assets/4bb9d5bd-b3b7-4645-9426-df524e11ce93)

2. I've modified https://github.com/hovsg/HOV-SG/blob/main/data/habitat/gen_hm3dsem_from_poses.py#L84 and https://github.com/hovsg/HOV-SG/blob/main/data/habitat/gen_hm3dsem_from_poses.py#L85 from "quite" to "verbose" in order to get more detailed error message.

3. After that, I executed the following command:

```bash
python data/habitat/gen_hm3dsem_from_poses.py --dataset_dir <hm3dsem_dir> --save_dir data/hm3dsem_walks/
```

### Results:

Script have been executed for some time, but in the end, it failed with the following error:
```bash
agent_state: position [13.376938    0.09943584  0.18875428] rotation quaternion(1, 0, 0, 0)
saving frame 3584/3586: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3585/3585 [03:48<00:00, 15.68it/s]
[15:40:40:663264]:[Debug]:[Physics] PhysicsManager.cpp(57)::~PhysicsManager : Deconstructing PhysicsManager
[15:40:40:663379]:[Debug]:[Scene] SceneManager.h(21)::~SceneManager : Deconstructing SceneManager
[15:40:40:663397]:[Debug]:[Scene] SceneGraph.h(25)::~SceneGraph : Deconstructing SceneGraph
[15:40:40:663973]:[Debug]:[Sensor] Sensor.cpp(68)::~Sensor : Deconstructing Sensor
[15:40:40:664186]:[Debug]:[Sensor] Sensor.cpp(68)::~Sensor : Deconstructing Sensor
[15:40:40:664413]:[Debug]:[Sensor] Sensor.cpp(68)::~Sensor : Deconstructing Sensor
[15:40:40:664608]:[Debug]:[Sensor] Sensor.cpp(68)::~Sensor : Deconstructing Sensor
[15:40:40:664647]:[Debug]:[Scene] SceneGraph.h(25)::~SceneGraph : Deconstructing SceneGraph
[15:40:40:667051]:[Debug]:[Scene] SemanticScene.h(53)::~SemanticScene : Deconstructing SemanticScene
[15:40:40:668724]:[Debug]:[Gfx] Renderer.cpp(71)::~Impl : Deconstructing Renderer
[15:40:40:668769]:[Debug]:[Gfx] WindowlessContext.h(17)::~WindowlessContext : Deconstructing WindowlessContext
[15:40:40:767203]:[Debug]:[Sim] Simulator.cpp(61)::~Simulator : Deconstructing Simulator
[15:40:40:767945]:[Debug]:[Physics] PhysicsManager.cpp(57)::~PhysicsManager : Deconstructing PhysicsManager
[15:40:40:767990]:[Debug]:[Scene] SceneManager.h(21)::~SceneManager : Deconstructing SceneManager
[15:40:40:768002]:[Debug]:[Scene] SceneGraph.h(25)::~SceneGraph : Deconstructing SceneGraph
GL::Context::current(): no current context
[1]    235042 IOT instruction (core dumped)  python data/habitat/gen_hm3dsem_from_poses.py --dataset_dir  --save_dir 
```

Generated data looks like this:
<img width="768" alt="Screenshot 2024-07-22 at 16 43 09" src="https://github.com/user-attachments/assets/f1df7b41-fcdf-4ae7-9173-c2bca0c57d9e">

### Questions

Have you seen similar problems? Could you help figure out what might be the root cause of the issue? Thanks.

By the way, it seems like it might be a habitat problem. Could you specify the version of habitat you are using?

**Comment by ADetilie** [comment]
Also, when I am trying to run "create_graph.py" script on any other folder, rather than `00824-Dd4bFSTQ8gi` (since it is empty for me), I am getting the following error:

```bash
(hovsg) ➜  HOV-SG git:(main) ✗ python application/create_graph.py main.dataset=hm3dsem main.dataset_path=data/hm3dsem_walks/val/00829-QaLdnwvtxbs main.save_path=data/scene_graphs/00829-QaLdnwvtxbs
/home/ncdev/miniforge3/envs/hovsg/lib/python3.9/site-packages/scipy/__init__.py:155: UserWarning: A NumPy version >=1.18.5 and <1.26.0 is required for this version of SciPy (detected version 1.26.4
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
[2024-07-22 17:00:38,332][root][INFO] - Loaded ViT-H-14 model config.
[2024-07-22 17:00:46,082][root][INFO] - Loading pretrained ViT-H-14 weights (checkpoints/laion2b_s32b_b79k.bin).
Creating RGB-D point cloud:   0%|                                                                                                                                                   | 0/181 [00:00<?, ?it/s]
Error executing job with overrides: ['main.dataset=hm3dsem', 'main.dataset_path=data/hm3dsem_walks/val/00829-QaLdnwvtxbs', 'main.save_path=data/scene_graphs/00829-QaLdnwvtxbs']
Traceback (most recent call last):
  File "/home/ncdev/repos/HOV-SG/application/create_graph.py", line 39, in <module>
    main()
  File "/home/ncdev/miniforge3/envs/hovsg/lib/python3.9/site-packages/hydra/main.py", line 94, in decorated_main
    _run_hydra(
  File "/home/ncdev/miniforge3/envs/hovsg/lib/python3.9/site-packages/hydra/_internal/utils.py", line 394, in _run_hydra
    _run_app(
  File "/home/ncdev/miniforge3/envs/hovsg/lib/python3.9/site-packages/hydra/_internal/utils.py", line 457, in _run_app
    run_and_report(
  File "/home/ncdev/miniforge3/envs/hovsg/lib/python3.9/site-packages/hydra/_internal/utils.py", line 223, in run_and_report
    raise ex
  File "/home/ncdev/miniforge3/envs/hovsg/lib/python3.9/site-packages/hydra/_internal/utils.py", line 220, in run_and_report
    return func()
  File "/home/ncdev/miniforge3/envs/hovsg/lib/python3.9/site-packages/hydra/_internal/utils.py", line 458, in <lambda>
    lambda: hydra.run(
  File "/home/ncdev/miniforge3/envs/hovsg/lib/python3.9/site-packages/hydra/_internal/hydra.py", line 132, in run
    _ = ret.return_value
  File "/home/ncdev/miniforge3/envs/hovsg/lib/python3.9/site-packages/hydra/core/utils.py", line 260, in return_value
    raise self._return_value
  File "/home/ncdev/miniforge3/envs/hovsg/lib/python3.9/site-packages/hydra/core/utils.py", line 186, in run_job
    ret.return_value = task_function(task_cfg)
  File "/home/ncdev/repos/HOV-SG/application/create_graph.py", line 19, in main
    hovsg.create_feature_map() # create feature map
  File "/home/ncdev/repos/HOV-SG/hovsg/graph/graph.py", line 154, in create_feature_map
    rgb_image, depth_image, pose, _, depth_intrinsics = self.dataset[i]
  File "/home/ncdev/repos/HOV-SG/hovsg/dataloader/hm3dsem.py", line 52, in __getitem__
    pose = self._load_pose(pose_path)
  File "/home/ncdev/repos/HOV-SG/hovsg/dataloader/hm3dsem.py", line 123, in _load_pose
    transformation_matrix = np.array(values).reshape((4, 4))
ValueError: cannot reshape array of size 1 into shape (4,4)
```

**Comment by abwerby** [comment]
Hello, 
I have not encountered this issue before it looks like a habitat sim problem. 
Does habitat-sim work with you in general?

**Comment by ADetilie** [comment]
When I run Habitat with a command like:
```bash
habitat-viewer mL8ThkuaVTM.glb
```
I don't encounter any problems. 

However, when I run the script, I get the following error at the end: `GL::Context::current(): no current context`. 
This error doesn't affect data generation tho; the data is still written to disk.

**Comment by ADetilie** [comment]
@abwerby but I've added to the `data/habitat/gen_hm3dsem_from_poses.py` script the following line (line 141):
```python
        sim.close()
 ```
 after the `for pose in pbar:` loop.
 It seems like this fiexed the problem.
 
 See the code snippet:
 ```python
         for pose in pbar:
            pbar.set_description(f"saving frame {steps}/{len(poses_list) + 1}")
            agent = sim.get_agent(0)
            agent_state = agent.get_state()
            agent_state.sensor_states["color_sensor"].position = pose[:3]
            agent_state.sensor_states["color_sensor"].rotation = pose[3:]
            agent_state.sensor_states["depth_sensor"].position = pose[:3]
            agent_state.sensor_states["depth_sensor"].rotation = pose[3:]
            agent_state.sensor_states["semantic"].position = pose[:3]
            agent_state.sensor_states["semantic"].rotation = pose[3:]
            agent.set_state(agent_state, reset_sensors=True, infer_sensor_states=False)
            obs = sim.get_sensor_observations(0)
            rgb = obs["color_sensor"]
            depth = obs["depth_sensor"]
            depth = ((depth / 10) * 255).astype(np.uint8)
            semantic = obs["semantic"]
            # cv2.imshow("rgb", rgb)
            # cv2.imshow("depth", depth)
            # cv2.waitKey()
            save_obs(save_dir, sim_settings, obs, pose, steps)
            steps += 1
        sim.close()
 ```

**Comment by abwerby** [comment]
thank you for sharing your solution.
